operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -543,10 +548,13 @@ func (c *cesManagerIdentity) InsertCEPInCache(cep *cilium_v2.CoreCiliumEndpoint,
 		if c.cesToIdentity[cesName] != cep.IdentityID {
 			c.RemoveCEPFromCache(GetCEPNameFromCCEP(cep, ns), DelayedCESSyncTime)
 		} else {
-			ces := c.desiredCESs.getCESTrackerOnly(cesName)
-			// add a cep into the ces
-			c.addCEPtoCES(cep, ces)
-			return cesName
+			if ces, ok := c.desiredCESs.getCESTracker(cesName); ok {
+				// add a cep into the ces
+				c.addCEPtoCES(cep, ces)
+				return cesName
+			} else {
+				log.Infof("Attempted to InsertCEPInCache non-existent cesName : %v, skipping.", cesName)
```

Comment:
Is this an error condition? I wonder whether we should be consuming such cases at this sort of level or attempting to forward an error up the stack to hint that the CES cannot be properly implemneted.